src/benchq/resource_estimators/graph_estimator.py

```diff
@@ -62,440 +66,119 @@ def __init__(
-                if this_transpilation_failure_tolerance < 1e-25:
-                    warnings.warn(
-                        "Synthesis tolerance low. Smaller problem size recommended.",
-                        RuntimeWarning,
-                    )
-                this_transpilation_failure_tolerance /= 10
-            else:
-                break
+                    + compiled_implementation.program.n_rotation_gates
+                    * n_t_gates_per_rotation
+                )
+            per_t_gate_failure_tolerance = total_t_gate_failure_tolerance / n_t_states
+            # Find minimal space or time factory satisfying error rate
+            magic_state_factory = find_optimal_factory(
+                per_t_gate_failure_tolerance,
+                magic_state_factory_iterator,
+                self.optimization,
+            )
 
-        # Get logical architecture resource info
-        # Generate spatial layout of data, bus, and magic state factories
+        # Find optimal layout by minimizing code distance
         logical_architecture_resource_info = (
-            self.get_bus_architecture_resource_breakdown(
+            logical_architecture_model.generate_minimal_code_distance_resources(
                 compiled_implementation.program,
-                data_and_bus_code_distance,
-                magic_state_factory,
+                self.optimization,
                 n_t_gates_per_rotation,
+                total_qec_failure_tolerance,
+                magic_state_factory,
+                hw_model,
             )
         )
 
-        # Generate temporal resource accounting
-        time_allocation = self.get_cycle_allocation(
-            compiled_implementation.program,
-            magic_state_factory.distillation_time_in_cycles,
-            n_t_gates_per_rotation,
-            data_and_bus_code_distance,
-        )
-        logical_architecture_resource_info.cycle_allocation = time_allocation
-
-        # Compute total failure rate
-        num_logical_qubits = (
-            logical_architecture_resource_info.num_logical_data_qubits
-            + logical_architecture_resource_info.num_logical_bus_qubits
-        )
-
-        num_cycles = logical_architecture_resource_info.cycle_allocation.total
-
-        st_volume_in_logical_qubit_tocks = (
-            num_logical_qubits * num_cycles / data_and_bus_code_distance
-        )
-
-        total_logical_error_rate = get_total_logical_failure_rate(
-            hw_model,
-            st_volume_in_logical_qubit_tocks,
-            data_and_bus_code_distance,
-        )
-
-        distillation_error_rate = (
-            magic_state_factory.distilled_magic_state_error_rate
-            * compiled_implementation.program.get_n_t_gates_after_transpilation(
-                this_transpilation_failure_tolerance
-            )
-        )
-
-        this_hardware_failure_rate = distillation_error_rate + total_logical_error_rate
+        # Populate resource info
```

Comment:
Good point. Currently the AbstractLogicalResourceInfo object is populated by the compile_and_estimate function [here](https://github.com/zapatacomputing/benchq/blob/98c59f9ab9f2df6a8637536899cdd50b243e4d6d/src/benchq/resource_estimators/graph_estimator.py#L513). However, I think we should aim to make the logic more clear. For example, it might be better to initialize a resource_info object at the beginning of or before compile_and_estimate (at which point we could already populate the AbstractLogicalResourceInfo object), and then the compile_and_estimate function could be responsible for populating the remaining fields of the resource info object.